src/main/java/de/tum/cit/aet/artemis/communication/repository/ConversationMessageRepository.java

```diff
@@ -43,56 +29,13 @@ public interface ConversationMessageRepository extends ArtemisJpaRepository<Post
 
     Logger log = LoggerFactory.getLogger(ConversationMessageRepository.class);
 
-    /**
-     * Configures the search specifications based on the provided filter criteria.
-     *
-     * @param specification     The existing specification to be configured.
-     * @param postContextFilter Filtering and sorting properties for post objects.
-     * @param userId            The id of the user for which the messages should be returned.
-     * @return A Specification object configured with search criteria.
-     */
-    private Specification<Post> configureSearchSpecification(Specification<Post> specification, PostContextFilterDTO postContextFilter, long userId) {
-        return specification
-        // @formatter:off
-            .and(getSearchTextAndAuthorSpecification(postContextFilter.searchText(), postContextFilter.authorIds()))
-            .and(getCourseWideChannelsSpecification(Boolean.TRUE.equals(postContextFilter.filterToCourseWide()), postContextFilter.courseId()))
-            .and(getAnsweredOrReactedSpecification(Boolean.TRUE.equals(postContextFilter.filterToAnsweredOrReacted()), userId))
-            .and(getUnresolvedSpecification(Boolean.TRUE.equals(postContextFilter.filterToUnresolved())))
-            .and(getPinnedSpecification(Boolean.TRUE.equals(postContextFilter.pinnedOnly())))
-            .and(getSortSpecification(true, postContextFilter.postSortCriterion(), postContextFilter.sortingOrder()));
-            // @formatter:on
-    }
-
-    /**
-     * Generates SQL Query via specifications to find and sort Messages
-     *
-     * @param postContextFilter filtering and sorting properties for post objects
-     * @param pageable          paging object which contains the page number and number of records to fetch
-     * @param userId            the id of the user for which the messages should be returned
-     * @return returns a Page of Messages
-     */
-    default Page<Post> findMessages(PostContextFilterDTO postContextFilter, Pageable pageable, long userId) {
-        var specification = getConversationsSpecification(postContextFilter.conversationIds());
-        specification = configureSearchSpecification(specification, postContextFilter, userId);
-        // Fetch all necessary attributes to avoid lazy loading (even though relations are defined as EAGER in the domain class, specification queries do not respect this)
-        return findPostsWithSpecification(pageable, specification);
-    }
+    // Eagerly fetch via query/graph
+    // Fetch only Post with direct relationships (author, conversation)—not collections.
+    @EntityGraph(attributePaths = { "author", "conversation" })
+    List<Post> findByIdIn(List<Long> postIds);
```

Comment:
From my perspective the comments do not add much value here and there is the risk that they will not be updated and start "lying"
```suggestion
    @EntityGraph(attributePaths = { "author", "conversation" })
    List<Post> findByIdIn(List<Long> postIds);
```